addons/edi/models/edi_gateway.py

```diff
@@ -177,9 +177,12 @@ class EdiGateway(models.Model):
     @api.depends("model_id")
     def _compute_can_initiate(self):
         """Compute ability to initiate connections"""
-        for gw in self.filtered("model_id"):
-            Model = self.env[gw.model_id.model]
-            gw.can_initiate = hasattr(Model, "connect")
+        for gw in self:
```

Comment:
Q: If the error is that there is not a value on some of the fields, could we keep the code the same but set `default=False` on the field?

Reply:
Is a computed field, generally on computed fields its not a good idea to have default.
On computed fields we need to return value on every recordset thats why we should not filter and in fact self cannot be filtered which was the place that raising error.

Hope it makes sense